zcash_client_backend/src/data_api/wallet.rs

```diff
@@ -521,14 +521,59 @@ where
 /// to fall back to the transparent receiver until full Orchard support is implemented.
 #[allow(clippy::too_many_arguments)]
 #[allow(clippy::type_complexity)]
-pub fn create_proposed_transaction<DbT, ParamsT, InputsErrT, FeeRuleT, N>(
+pub fn create_proposed_transactions<DbT, ParamsT, InputsErrT, FeeRuleT, N>(
     wallet_db: &mut DbT,
     params: &ParamsT,
     spend_prover: &impl SpendProver,
     output_prover: &impl OutputProver,
     usk: &UnifiedSpendingKey,
     ovk_policy: OvkPolicy,
     proposal: &Proposal<FeeRuleT, N>,
+) -> Result<
+    Vec<TxId>,
+    Error<
+        <DbT as WalletRead>::Error,
+        <DbT as WalletCommitmentTrees>::Error,
+        InputsErrT,
+        FeeRuleT::Error,
+    >,
+>
+where
+    DbT: WalletWrite + WalletCommitmentTrees,
+    ParamsT: consensus::Parameters + Clone,
+    FeeRuleT: FeeRule,
+{
+    proposal
+        .steps()
+        .iter()
+        .map(|step| {
+            create_proposed_transaction(
```

Comment:
This will need to be altered once the `Proposal` protobuf is fixed to allow chained transactions, because as-implemented the proposal steps are not steps, but are completely independent set elements that can be created in arbitrary order.